### PR TITLE
Rework URL Encoding

### LINKED ISSUE #779: Rework URL Encoding
- Replace UrlEncode uses with EscapeDataString
- Replace Extension Function uses with EscapeDataString

Why? Comply with RFC2396

https://stackoverflow.com/questions/3572173/server-urlencode-vs-uri-escapedatastring

replace https://docs.microsoft.com/en-us/dotnet/api/system.net.webutility.urlencode

with https://docs.microsoft.com/en-us/dotnet/api/system.uri.escapedatastring?view=net-6.0

[AB#2369](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2369)

### PR DESCRIPTION
**Pull request by foux**
#### Database Migration
NO

#### Description
Change URL encoding to properly encode all non-ascii characters

#### Todos
- [ ] Tests

#### Issues Fixed or Closed by this PR

* Fixes #779

### CODE REVIEW

**Comment by foux** [comment]
@Qstick not sure if I should change in the indexers definitions, or if those are generated

File: src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs [hunk 1]
```diff
@@ -1013,7 +1013,7 @@ private IEnumerable<IndexerRequest> GetRequest(Dictionary<string, object> variab
 
                 // build search URL
                 // HttpUtility.UrlPathEncode seems to only encode spaces, we use UrlEncode and replace + with %20 as a workaround
-                var searchUrl = ResolvePath(ApplyGoTemplateText(searchPath.Path, variables, WebUtility.UrlEncode).Replace("+", "%20")).AbsoluteUri;
+                var searchUrl = ResolvePath(ApplyGoTemplateText(searchPath.Path, variables, Uri.EscapeDataString).Replace("+", "%20")).AbsoluteUri;
```
Comment (Qstick) [inline src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs:1016]: `EscapeDataString` replaces + with %20 already right? So we can drop the extra replace and the note above

Comment (foux) [reply src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs]: Off course. Changed

**Comment by foux** [comment]
OK, so this PR is pure junk, it totally destroys the URLs. For example : 
`http://localhost:9696/1/download\?apikey\=2475bb5b008d4522834ee0494806b59f\&link\=cmVIZGp0RzFPazF4bWpMRVpCUkQxalVpcEhrUUsyOXk2TkVoaENieXpmdDFsbEJBaWMxcmpPYzk4SzRPK1VNSW1DSHRZY3RiVmVINVRmK0VOdTh1UlpEMHpNdzZJekJiNmpZcXdaSlVYSHNGTSthdE1abDJuTlRWVHcrc0tEQ0M\&file\=Andrée A  Michaud - Tempêtes - MP3 64 Kb%2Fs`

**Pull request merged**